x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCachePlugin.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.blobcache;
+
+import org.elasticsearch.blobcache.shared.SharedBlobCacheService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.plugins.ExtensiblePlugin;
+import org.elasticsearch.plugins.Plugin;
+
+import java.util.List;
+
+public class BlobCachePlugin extends Plugin implements ExtensiblePlugin {
```

Comment:
Making this `ExtensiblePlugin` and using it from `SearchableSnapshots` was the only way I found to make the classloading work out such that the `preallocate` permissions and the settings are guaranteed to get set up before using them in `SearchableSnapshots`.